### ISSUE TITLE
Switch to using reference-style links

### ISSUE THREAD

**Issue by flexibeast** [void-linux/void-docs#286]
@jeffayle suggested [here](https://github.com/void-linux/void-docs/pull/248#discussion_r425310264) that we use reference-style links in the Handbook when linking to man pages. i like this idea, particularly as we factor out common information and increasingly link to page fragments: if all man page links can simply have the format `[ls(1)]`, we won't have to worry about users being directed to a fragment which contains a non-linked man page reference.

**Comment by ericonr** [comment]
I love this idea! Some refactoring would be needed, of course :p

**Comment by jeffayle** [comment]
My recommendation:
- man pages should always use reference style links
- other links should use a reference if the url is too long to be reflowed in a nice way (eg https://github.com/void-linux/void-docs/blame/master/src/config/graphical-session/xorg.md#L40 ) or if the same url is linked in more than one place.
- man pages should use the form `[linkref]` whereas other links should use the form `[link text][linkref]`
- it might be worth pointing to [the spec](https://spec.commonmark.org/0.29/#full-reference-link)  which has a lot of examples

vmdfmt rewrites reference style links as inline links, so that's going to be an issue.

**Comment by hippi777** [comment]
maybe that "spec" link could be included in the docs somewhere around contributing

**Comment by flexibeast** [comment]
@hippi777: Yeah, it would probably be good to reference it from the style guide.

**Comment by flexibeast** [comment]
@jeffayle: Oh, so we can't just start using reference links without having to modify `vmdfmt` first? i'd been assuming `vmdfmt` could already handle this.

**Comment by flexibeast** [comment]
Upon further thought, even though i can understand why it might be useful to point to the CommonMark spec as a general guide to Markdown, i'm not sure we should actually do so, since we use Versioned Markdown, not CommonMark - i wouldn't want people to assume that they can use anything described by the latter spec. That said, it could certainly be used as a basis for adding reference-style links to Versioned Markdown.

**Comment by jeffayle** [comment]
@flexibeast yeah, I assumed vmdfmt would just leave them alone if it hadn't been written to handle it. I might take a look at vmdfmt, but I've never programmed in go before, and don't have a whole lot of experience with parsers
